src/org/sipdroid/sipua/SipdroidEngine.java

```diff
@@ -529,6 +568,11 @@ public void halt() { // modified
 				sip_providers[i].halt();
 			i++;
 		}
+		i=0;
+		for (PublishAgent pa : pas) {

```

Comment:
why this weird way of doing things instead of calling the function on the `pa` instance?